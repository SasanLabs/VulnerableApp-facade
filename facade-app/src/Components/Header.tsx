import React from "react";
import VulnerableAppLogo from "../images/Logo.png";

import {
  Header as RSuiteHeader,
  Navbar as RSuiteNavBar,
  Nav as RSuiteNav,
  Icon as RSuiteIcon,
  Dropdown as RSuiteDropDown,
} from "rsuite";
import { Props } from "../interface/Props";

import "../styles/Header.css";

export default class Header extends React.Component<Props, {}> {
  render() {
    const { globalState, setGlobalState } = this.props;
    return (
      <RSuiteHeader>
        <RSuiteNavBar appearance="inverse">
          <RSuiteNavBar.Header>
            <img
              src={VulnerableAppLogo}
              width="55"
              height="55"
              alt="vulnerable app logo"
            />
          </RSuiteNavBar.Header>
          <RSuiteNavBar.Body>
            <RSuiteNav>
              <RSuiteNav.Item
                onSelect={() =>
                  setGlobalState({
                    ...globalState,
                    activateHomePage: true,
                    activateAboutUsPage: false,
                  })
                }
              >
                <b>Owasp VulnerableApp-Facade</b>
              </RSuiteNav.Item>
            </RSuiteNav>
            <RSuiteNav pullRight>
              <RSuiteNav.Item
                onSelect={() =>
                  setGlobalState({
                    ...globalState,
                    activateHomePage: true,
                    activateAboutUsPage: false,
                  })
                }
                icon={<RSuiteIcon icon="home" role={"img"} />}
              >
                Home
              </RSuiteNav.Item>
              <RSuiteNav.Item
                onSelect={() =>
                  setGlobalState({
                    ...globalState,
                    activateHomePage: false,
                    activateAboutUsPage: true,
                  })
                }
              >
                About Us
              </RSuiteNav.Item>
              <RSuiteDropDown title="Scanners">
                <RSuiteDropDown.Item title="Dynamic Application Security Testing">
                  DAST
                </RSuiteDropDown.Item>
                <RSuiteDropDown.Item title="Static Application Security Testing">
                  SAST
                </RSuiteDropDown.Item>
              </RSuiteDropDown>
              <a href="https://github.com/SasanLabs/VulnerableApp-facade">
                <RSuiteNav.Item
                  icon={<RSuiteIcon icon="github" role={"img"} />}
                >
                  Github
                </RSuiteNav.Item>
              </a>
              <RSuiteDropDown title="Projects by SasanLabs">
                <a href="https://github.com/SasanLabs/VulnerableApp">
                  <RSuiteDropDown.Item
                    icon={<RSuiteIcon icon="github" role={"img"} />}
                  >
                    Owasp VulnerableApp
                  </RSuiteDropDown.Item>
                </a>
                <a href="https://github.com/SasanLabs/owasp-zap-jwt-addon">
                  {" "}
                  <RSuiteDropDown.Item
                    icon={<RSuiteIcon icon="github" role={"img"} />}
                  >
                    ZAP JWT Addon
                  </RSuiteDropDown.Item>
                </a>
                <a href="https://github.com/SasanLabs/owasp-zap-fileupload-addon">
                  <RSuiteDropDown.Item
                    icon={<RSuiteIcon icon="github" role={"img"} />}
                  >
                    ZAP FileUpload Addon
                  </RSuiteDropDown.Item>
                </a>
              </RSuiteDropDown>
            </RSuiteNav>
          </RSuiteNavBar.Body>
        </RSuiteNavBar>
      </RSuiteHeader>
    );
  }
}
