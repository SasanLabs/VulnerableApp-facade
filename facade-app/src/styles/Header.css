/* scanner dropdown style */
.rs-dropdown-menu {
  width: 100%;
}
