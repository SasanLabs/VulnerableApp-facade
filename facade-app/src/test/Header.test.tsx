import React from "react";
import { render, screen } from "@testing-library/react";

import Header from "../Components/Header";

describe("Header", () => {
  it("renders correctly", () => {
    const mock = () => jest.fn();
    const state = {
      isSuccessfullyLoaded: false,
      activateAboutUsPage: false,
      activateHomePage: true,
      showHints: false,
    };
    const { container } = render(
      <Header setGlobalState={mock} globalState={state} />
    );

    expect(container).toMatchSnapshot();
  });

  it("renders header image", () => {
    const mock = () => jest.fn();
    const state = {
      isSuccessfullyLoaded: false,
      activateAboutUsPage: false,
      activateHomePage: true,
      showHints: false,
    };
    render(<Header setGlobalState={mock} globalState={state} />);

    const image = screen.getAllByRole("img")[0];

    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute("alt", "vulnerable app logo");
  });

  it("renders first nav item", () => {
    const mock = () => jest.fn();
    const state = {
      isSuccessfullyLoaded: false,
      activateAboutUsPage: false,
      activateHomePage: true,
      showHints: false,
    };
    render(<Header setGlobalState={mock} globalState={state} />);

    const item = screen.getByText("Owasp VulnerableApp-Facade");

    expect(item).toBeInTheDocument();
  });

  it("renders second nav item", () => {
    const mock = () => jest.fn();
    const state = {
      isSuccessfullyLoaded: false,
      activateAboutUsPage: false,
      activateHomePage: true,
      showHints: false,
    };
    render(<Header setGlobalState={mock} globalState={state} />);

    const homeIcon = screen.getAllByRole("img")[1];
    const text = screen.getByText("Home");

    expect(homeIcon).toBeInTheDocument();
    expect(text).toBeInTheDocument();
  });

  it("renders third nav item", () => {
    const mock = () => jest.fn();
    const state = {
      isSuccessfullyLoaded: false,
      activateAboutUsPage: false,
      activateHomePage: true,
      showHints: false,
    };
    render(<Header setGlobalState={mock} globalState={state} />);

    const text = screen.getByText("About Us");

    expect(text).toBeInTheDocument();
  });

  it("renders Github link", () => {
    const mock = () => jest.fn();
    const state = {
      isSuccessfullyLoaded: false,
      activateAboutUsPage: false,
      activateHomePage: true,
      showHints: false,
    };
    render(<Header setGlobalState={mock} globalState={state} />);

    const link = screen.getAllByRole("link")[0];
    const githubIcon = screen.getAllByRole("img")[2];
    const text = screen.getByText("Github");

    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute(
      "href",
      "https://github.com/SasanLabs/VulnerableApp-facade"
    );
    expect(githubIcon).toBeInTheDocument();
    expect(text).toBeInTheDocument();
  });

  describe("dropdown", () => {
    it("renders dropdown", () => {
      const mock = () => jest.fn();
      const state = {
        isSuccessfullyLoaded: false,
        activateAboutUsPage: false,
        activateHomePage: true,
        showHints: false,
      };
      render(<Header setGlobalState={mock} globalState={state} />);

      const dropDownTitle = screen.getByText("Projects by SasanLabs");

      expect(dropDownTitle).toBeInTheDocument();
    });

    it("renders first item", () => {
      const mock = () => jest.fn();
      const state = {
        isSuccessfullyLoaded: false,
        activateAboutUsPage: false,
        activateHomePage: true,
        showHints: false,
      };
      render(<Header setGlobalState={mock} globalState={state} />);

      const owaspLink = screen.getAllByRole("link")[1];
      const owaspIcon = screen.getAllByRole("img")[3];
      const text = screen.getByText("Owasp VulnerableApp");

      expect(owaspLink).toBeInTheDocument();
      expect(owaspLink).toHaveAttribute(
        "href",
        "https://github.com/SasanLabs/VulnerableApp"
      );
      expect(owaspIcon).toBeInTheDocument();
      expect(text).toBeInTheDocument();
    });

    it("renders second item", () => {
      const mock = () => jest.fn();
      const state = {
        isSuccessfullyLoaded: false,
        activateAboutUsPage: false,
        activateHomePage: true,
        showHints: false,
      };
      render(<Header setGlobalState={mock} globalState={state} />);

      const zapLink = screen.getAllByRole("link")[2];
      const zapIcon = screen.getAllByRole("img")[4];
      const text = screen.getByText("ZAP JWT Addon");

      expect(zapLink).toBeInTheDocument();
      expect(zapLink).toHaveAttribute(
        "href",
        "https://github.com/SasanLabs/owasp-zap-jwt-addon"
      );
      expect(zapIcon).toBeInTheDocument();
      expect(text).toBeInTheDocument();
    });

    it("renders third item", () => {
      const mock = () => jest.fn();
      const state = {
        isSuccessfullyLoaded: false,
        activateAboutUsPage: false,
        activateHomePage: true,
        showHints: false,
      };
      render(<Header setGlobalState={mock} globalState={state} />);

      const zapLink = screen.getAllByRole("link")[3];
      const zapIcon = screen.getAllByRole("img")[5];
      const text = screen.getByText("ZAP FileUpload Addon");

      expect(zapLink).toBeInTheDocument();
      expect(zapLink).toHaveAttribute(
        "href",
        "https://github.com/SasanLabs/owasp-zap-fileupload-addon"
      );
      expect(zapIcon).toBeInTheDocument();
      expect(text).toBeInTheDocument();
    });
  });

  describe("scanner dropdown", () => {
    it("renders scanner dropdown", () => {
      const mock = () => jest.fn();
      const state = {
        isSuccessfullyLoaded: false,
        activateAboutUsPage: false,
        activateHomePage: true,
        showHints: false,
      };
      render(<Header setGlobalState={mock} globalState={state} />);

      const dropDownTitle = screen.getByText("Scanners");

      expect(dropDownTitle).toBeInTheDocument();
    });

    it("renders first item", () => {
      const mock = () => jest.fn();
      const state = {
        isSuccessfullyLoaded: false,
        activateAboutUsPage: false,
        activateHomePage: true,
        showHints: false,
      };
      render(<Header setGlobalState={mock} globalState={state} />);
      const text = screen.getByText("DAST");
      expect(text).toBeInTheDocument();
    });

    it("renders second item", () => {
      const mock = () => jest.fn();
      const state = {
        isSuccessfullyLoaded: false,
        activateAboutUsPage: false,
        activateHomePage: true,
        showHints: false,
      };
      render(<Header setGlobalState={mock} globalState={state}></Header>);
      const text = screen.getByText("SAST");
      expect(text).toBeInTheDocument();
    });

    it("should have tooltip for DAST", () => {
      const mock = () => jest.fn();
      const state = {
        isSuccessfullyLoaded: false,
        activateAboutUsPage: false,
        activateHomePage: true,
        showHints: false,
      };
      const dast = "Dynamic Application Security Testing";
      render(<Header setGlobalState={mock} globalState={state}></Header>);
      expect(screen.getByTitle(dast)).toBeInTheDocument();
    });

    it("should have tooltip for SAST", () => {
      const mock = () => jest.fn();
      const state = {
        isSuccessfullyLoaded: false,
        activateAboutUsPage: false,
        activateHomePage: true,
        showHints: false,
      };
      const sast = "Static Application Security Testing";
      render(<Header setGlobalState={mock} globalState={state}></Header>);
      expect(screen.getByTitle(sast)).toBeInTheDocument();
    });
  });
});
