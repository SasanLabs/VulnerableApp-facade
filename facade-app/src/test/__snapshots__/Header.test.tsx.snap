// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div>
  <div
    class="rs-header"
  >
    <div
      class="rs-navbar rs-navbar-inverse"
      role="navigation"
    >
      <div
        class="rs-navbar-header"
      >
        <img
          alt="vulnerable app logo"
          height="55"
          src="Logo.png"
          width="55"
        />
      </div>
      <div
        class="rs-navbar-body"
      >
        <div
          class="rs-nav rs-nav-default rs-navbar-nav rs-nav-horizontal"
        >
          <ul>
            <li
              class="rs-nav-item"
            >
              <a
                class="rs-nav-item-content"
                role="button"
                tabindex="0"
              >
                <b>
                  Owasp VulnerableApp-Facade
                </b>
                <span
                  class="rs-ripple-pond"
                >
                  <span
                    class="rs-ripple"
                  />
                </span>
              </a>
            </li>
          </ul>
        </div>
        <div
          class="rs-nav rs-nav-default rs-navbar-nav rs-navbar-right rs-nav-horizontal"
        >
          <ul>
            <li
              class="rs-nav-item"
            >
              <a
                class="rs-nav-item-content"
                role="button"
                tabindex="0"
              >
                <i
                  class="rs-icon rs-icon-home"
                  role="img"
                />
                Home
                <span
                  class="rs-ripple-pond"
                >
                  <span
                    class="rs-ripple"
                  />
                </span>
              </a>
            </li>
            <li
              class="rs-nav-item"
            >
              <a
                class="rs-nav-item-content"
                role="button"
                tabindex="0"
              >
                About Us
                <span
                  class="rs-ripple-pond"
                >
                  <span
                    class="rs-ripple"
                  />
                </span>
              </a>
            </li>
            <li
              class="rs-dropdown rs-dropdown-placement-bottom-start"
              role="menu"
            >
              <ul
                class="rs-dropdown-menu"
                role="menu"
              >
                <li
                  class="rs-dropdown-item"
                >
                  <a
                    class="rs-dropdown-item-content"
                    tabindex="-1"
                    title="Dynamic Application Security Testing"
                  >
                    DAST
                  </a>
                </li>
                <li
                  class="rs-dropdown-item"
                >
                  <a
                    class="rs-dropdown-item-content"
                    tabindex="-1"
                    title="Static Application Security Testing"
                  >
                    SAST
                  </a>
                </li>
              </ul>
              <a
                class="rs-btn rs-btn-subtle rs-dropdown-toggle"
                tabindex="0"
              >
                Scanners
                <span
                  class="rs-dropdown-toggle-caret"
                />
                <span
                  class="rs-ripple-pond"
                >
                  <span
                    class="rs-ripple"
                  />
                </span>
              </a>
            </li>
            <a
              href="https://github.com/SasanLabs/VulnerableApp-facade"
            >
              <li
                class="rs-nav-item"
              >
                <a
                  class="rs-nav-item-content"
                  role="button"
                  tabindex="0"
                >
                  <i
                    class="rs-icon rs-icon-github"
                    role="img"
                  />
                  Github
                  <span
                    class="rs-ripple-pond"
                  >
                    <span
                      class="rs-ripple"
                    />
                  </span>
                </a>
              </li>
            </a>
            <li
              class="rs-dropdown rs-dropdown-placement-bottom-start"
              role="menu"
            >
              <ul
                class="rs-dropdown-menu"
                role="menu"
              >
                <a
                  href="https://github.com/SasanLabs/VulnerableApp"
                >
                  <li
                    class="rs-dropdown-item rs-dropdown-item-with-icon"
                  >
                    <a
                      class="rs-dropdown-item-content"
                      tabindex="-1"
                    >
                      <i
                        class="rs-dropdown-item-menu-icon rs-icon rs-icon-github"
                        role="img"
                      />
                      Owasp VulnerableApp
                    </a>
                  </li>
                </a>
                <a
                  href="https://github.com/SasanLabs/owasp-zap-jwt-addon"
                >
                   
                  <li
                    class="rs-dropdown-item rs-dropdown-item-with-icon"
                  >
                    <a
                      class="rs-dropdown-item-content"
                      tabindex="-1"
                    >
                      <i
                        class="rs-dropdown-item-menu-icon rs-icon rs-icon-github"
                        role="img"
                      />
                      ZAP JWT Addon
                    </a>
                  </li>
                </a>
                <a
                  href="https://github.com/SasanLabs/owasp-zap-fileupload-addon"
                >
                  <li
                    class="rs-dropdown-item rs-dropdown-item-with-icon"
                  >
                    <a
                      class="rs-dropdown-item-content"
                      tabindex="-1"
                    >
                      <i
                        class="rs-dropdown-item-menu-icon rs-icon rs-icon-github"
                        role="img"
                      />
                      ZAP FileUpload Addon
                    </a>
                  </li>
                </a>
              </ul>
              <a
                class="rs-btn rs-btn-subtle rs-dropdown-toggle"
                tabindex="0"
              >
                Projects by SasanLabs
                <span
                  class="rs-dropdown-toggle-caret"
                />
                <span
                  class="rs-ripple-pond"
                >
                  <span
                    class="rs-ripple"
                  />
                </span>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
